### PR TITLE
Optimisation of imports

### LINKED ISSUE #137: Optimize the imports in the entire source code
In particular, the `reframe.settings` module is imported in:

* `reframe/frontend/executors/policies.py`
* `reframe/frontend/loader.py`
* `reframe/core/logging.py`
* `reframe/core/pipeline.py`

and is not used.

### PR DESCRIPTION
**Pull request by omlins**
Includes optimisation of .svg files

Fixes #137.

### CODE REVIEW

File: cscs-checks/apps/cpmd/cpmd_check.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import os
-import reframe.utility.sanity as sn
 
+import reframe.utility.sanity as sn
```
[the remark this answers is not in the document]
Comment (vkarak) [reply cscs-checks/apps/cpmd/cpmd_check.py]: Ok, I think that one is more [PEP8](https://www.python.org/dev/peps/pep-0008/#imports) compliant.

File: docs/_static/img/deferrable-evaluation.svg [hunk 2]
```diff
@@ -1,2 +1,3 @@
...
...7" y="12" fill="#000000" text-anchor="middle" font-size="12px" font-family="Helvetica">real dlen()</text></switch></g><path d="M 60 296 L 146 296 L 160 310 L 160 366 L 60 366 L 60 296 Z" fill="none" stroke="#4d4d4d" stroke-width="2" stroke-miterlimit="10" pointer-events="none"/><path d="M 146 296 L 146 310 L 160 310" fill="none" stroke="#4d4d4d" stroke-width="2" stroke-miterlimit="10" pointer-events="none"/><g transform="translate(62.5,297.5)"><switch><foreignObject style="overflow:visible;" pointer-events="all" width="95" height="54" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; font-size: 12px; font-family: Helvetica; color: rgb(0, 0, 0); line-height: 1.2; vertical-align: top; width: 95px; white-space: normal; word-wrap: normal;"><div xmlns="http://www.w3.org/1999/xhtml" style="display:inline-block;text-align:inherit;text-decoration:inherit;">Return value<div>is a deferred expression, so evaluate it, too.</div></div></div></foreignObject><text x="48" y="33" fill="#000000" text-anchor="middle" font-size="12px" font-family="Helvetica">[Not supported by viewer]</text></switch></g></g></svg>
```
Comment (vkarak) [inline docs/_static/img/deferrable-evaluation.svg]: I don't think this should be changed...

File: reframe/core/schedulers/registry.py [hunk 3]
```diff
@@ -30,5 +30,3 @@ def getscheduler(name):
 
 
 # Import the schedulers modules to trigger their registration
-import reframe.core.schedulers.local
-import reframe.core.schedulers.slurm
```
Comment (vkarak) [inline reframe/core/schedulers/registry.py]: These two are needed for registering the schedulers (see comment above).

File: reframe/core/launchers/registry.py [hunk 4]
```diff
@@ -77,5 +77,3 @@ def setup(self, partition, environ, **job_opts):
 
 
 # Import the launchers modules to trigger their registration
-import reframe.core.launchers.local
-import reframe.core.launchers.mpi
```
Comment (vkarak) [inline reframe/core/launchers/registry.py]: These two are needed for registering the launchers (see comment above).

File: unittests/resources/badchecks/notacheck.py [hunk 5]
```diff
@@ -1 +1 @@
-import _foo
```
Comment (vkarak) [inline unittests/resources/badchecks/notacheck.py]: This is needed to check a test having a Python file as resources and this files does an import that does not exist. We had a bug before with a test doing exactly that and crashing the framework.

File: tutorial/example1.py [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import os
-import reframe.utility.sanity as sn
 
+import reframe.utility.sanity as sn
```
Comment (vkarak) [inline tutorial/example1.py]: You should update also the `.rst` documentation files accordingly. Perhaps, it's better to use [this](https://goo.gl/u2mD9k) and avoid synchronizing the tutorial examples and the documentation every time. If it turns out not to be straightforward, you can just update the `.rst` files manually and we postpone this item.

File: cscs-checks/apps/cpmd/cpmd_check.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (omlins) [reply cscs-checks/apps/cpmd/cpmd_check.py]: This was done automatically. So I keep the change?

[the remark this answers is not in the document]
Comment (vkarak) [reply cscs-checks/apps/cpmd/cpmd_check.py]: Yes, you keep it, cos it is the correct interpretation of the PEP8 standard on imports.

File: tutorial/example1.py (hunk 6 above)
Comment (omlins) [reply tutorial/example1.py]: Done.

File: unittests/test_fields.py [hunk 7]
```diff
@@ -426,6 +427,7 @@ def __init__(self, value):
                           globals(), locals())
 
     def test_settings(self):
+        # This import is required even if Pycharm detects it as unneeded.
```
[the remark this answers is not in the document]
Comment (vkarak) [reply unittests/test_fields.py]: An acceptable alternative would be to put the `from reframe.settings import settings` inside the code executed by the `exec`. Perhaps sth like that:

```python
self.assertRaises(AttributeError, exec,
                  "from reframe.settings import settings\n"
                  "settings.checks_path = ['/foo']",
                  globals(), locals())
```

[the remark this answers is not in the document]
Comment (omlins) [reply unittests/test_fields.py]: Done.

File: reframe/frontend/loader.py [hunk 8]
```diff
@@ -5,19 +5,17 @@
...
 import os
-import logging
-import sys
+from importlib.machinery import SourceFileLoader
+
 import reframe.core.debug as debug
 import reframe.utility.os as os_ext
-
-from importlib.machinery import SourceFileLoader
+# Pycharm detects the import of ProgEnvironment wrongly as unneeded.
```
Comment (vkarak) [inline reframe/frontend/loader.py]: The source of this problem is further down the code. I'll do a PR into your branch to make PyCharm happy too!

**Comment by vkarak** [comment]
Looks great now. Thanks @omlins.

**Pull request merged**